MacOS Pipelines fail

### Issue Description

For some weeks our MacOS pipelines fail

### Minimal Reproducible Example

```python
https://github.com/shap/shap/actions/runs/12273890069/job/34245638917?pr=3917
```


### Traceback

```shell
def test_tf_deep_imbdb_transformers():
        # GH 3522
        transformers = pytest.importorskip("transformers")
    
        from shap import models
    
        # data from datasets imdb dataset
        short_data = ["I lov", "Worth", "its a", "STAR ", "First", "I had", "Isaac", "It ac", "Techn", "Hones"]
>       classifier = transformers.pipeline("sentiment-analysis", return_all_scores=True)

...

E       RuntimeError: MPS backend out of memory (MPS allocated: 1.01 GB, other allocations: 16.00 KB, max allowed: 7.93 GB). Tried to allocate 256 bytes on shared pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).
```


### Expected Behavior

should be running

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

latest

--- a/tests/explainers/test_deep.py
+++ b/tests/explainers/test_deep.py
@@ -1,5 +1,7 @@
 """Tests for the Deep explainer."""
 
+import platform
+
 import numpy as np
 import pandas as pd
 import pytest
@@ -241,6 +243,7 @@ def test_tf_keras_imdb_lstm(random_seed):
     np.testing.assert_allclose(sums, diff, atol=1e-02), "Sum of SHAP values does not match difference!"
 
 
+@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error on GH runners")
 def test_tf_deep_imbdb_transformers():
     # GH 3522
     transformers = pytest.importorskip("transformers")

--- a/tests/explainers/test_deep.py
+++ b/tests/explainers/test_deep.py
@@ -1,7 +1,5 @@
 """Tests for the Deep explainer."""
 
-import platform
-
 import numpy as np
 import pandas as pd
 import pytest
@@ -243,12 +241,14 @@ def test_tf_keras_imdb_lstm(random_seed):
     np.testing.assert_allclose(sums, diff, atol=1e-02), "Sum of SHAP values does not match difference!"
 
 
-@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error on GH runners")
-def test_tf_deep_imbdb_transformers():
+def test_tf_deep_imbdb_transformers(monkeypatch):
     # GH 3522
     transformers = pytest.importorskip("transformers")
 
     from shap import models
+
+    # Disable upper limit for memory allocations (see GH #3929)
+    monkeypatch.setenv("PYTORCH_MPS_HIGH_WATERMARK_RATIO", "0.0")
 
     # data from datasets imdb dataset
     short_data = ["I lov", "Worth", "its a", "STAR ", "First", "I had", "Isaac", "It ac", "Techn", "Hones"]

--- a/tests/explainers/test_deep.py
+++ b/tests/explainers/test_deep.py
@@ -243,7 +243,7 @@ def test_tf_keras_imdb_lstm(random_seed):
     np.testing.assert_allclose(sums, diff, atol=1e-02), "Sum of SHAP values does not match difference!"
 
 
-@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error on GH runners")
+@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error, see GH #3929")
 def test_tf_deep_imbdb_transformers():
     # GH 3522
     transformers = pytest.importorskip("transformers")

--- a/tests/explainers/test_deep.py
+++ b/tests/explainers/test_deep.py
@@ -1,5 +1,7 @@
 """Tests for the Deep explainer."""
 
+import platform
+
 import numpy as np
 import pandas as pd
 import pytest
@@ -241,14 +243,12 @@ def test_tf_keras_imdb_lstm(random_seed):
     np.testing.assert_allclose(sums, diff, atol=1e-02), "Sum of SHAP values does not match difference!"
 
 
-def test_tf_deep_imbdb_transformers(monkeypatch):
+@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error on GH runners")
+def test_tf_deep_imbdb_transformers():
     # GH 3522
     transformers = pytest.importorskip("transformers")
 
     from shap import models
-
-    # Disable upper limit for memory allocations (see GH #3929)
-    monkeypatch.setenv("PYTORCH_MPS_HIGH_WATERMARK_RATIO", "0.0")
 
     # data from datasets imdb dataset
     short_data = ["I lov", "Worth", "its a", "STAR ", "First", "I had", "Isaac", "It ac", "Techn", "Hones"]

--- a/tests/explainers/test_deep.py
+++ b/tests/explainers/test_deep.py
@@ -1,5 +1,6 @@
 """Tests for the Deep explainer."""
 
+import os
 import platform
 
 import numpy as np
@@ -243,7 +244,10 @@ def test_tf_keras_imdb_lstm(random_seed):
     np.testing.assert_allclose(sums, diff, atol=1e-02), "Sum of SHAP values does not match difference!"
 
 
-@pytest.mark.skipif(platform.system() == "Darwin", reason="Skipping on MacOS due to memory error, see GH #3929")
+@pytest.mark.skipif(
+    platform.system() == "Darwin" and os.getenv("GITHUB_ACTIONS") == "true",
+    reason="Skipping on GH MacOS runners due to memory error, see GH #3929",
+)
 def test_tf_deep_imbdb_transformers():
     # GH 3522
     transformers = pytest.importorskip("transformers")

